lib/srv/alpnproxy/local_proxy.go

```diff
@@ -71,6 +75,11 @@ type LocalProxyConfig struct {
 	Certs []tls.Certificate
 	// AWSCredentials are AWS Credentials used by LocalProxy for request's signature verification.
 	AWSCredentials *credentials.Credentials
+	// OnNewConnection is a callback triggered when a new downstream connection
+	// is accepted by the local proxy.
+	//
+	// Note that the callback blocks handling of the connection.
```

Comment:
I see this note but I just want to clarify that your intention was to block the LocalProxy `for {...}` loop entirely or just an individual connection? We are doing the former - but if that was your intention could you explain why?

Reply:
good question. I guess this note is inaccurate. In current use cases, I think we want to block the `LocalProxy.Start`, meaning when detecting a certain situation by this callback, there is no point in accepting another new connection until the situation is resolved. For the MFA flow, feel free to do anything that works for you.

Reply:
good question. I guess this note is inaccurate. In current use cases, I think we want to block the `LocalProxy.Start`, meaning when detecting a certain situation by this callback, there is no point in accepting another new connection until the situation is resolved. For the MFA flow, feel free to do anything that works for you.